lib/spack/spack/traverse.py

```diff
@@ -321,13 +321,13 @@ def traverse_depth_first_with_visitor(edges, visitor):
         if not visitor.accept(edge):
             continue
 
-        visitor.pre(edge)
+        # visitor.pre(edge)
```

Comment:
This is still work in progress. I tried to unify the cases using a visitor, as we talked about. Two things:
- Here I commented out `visitor.pre` and `visitor.post`. I assume it's fine to delete these calls? I don't see any visitor implementing them, or any place where this function is called
- For the `spack spec` default case, I need to recover the same ordering of specs. Any suggestion on ways to do that effectively is welcome
- I need to either remove or recover `cover=='paths'` I'd prefer remove, since I assume it's not very much used